utils/src/main/java/software/amazon/awssdk/utils/StringUtils.java

```diff
@@ -603,4 +620,142 @@ public static String uncapitalize(final String str) {
         }
         return new String(newCodePoints, 0, outOffset);
     }
+
+    /**
+     * This method appends a string to a string builder and collapses contiguous
+     * white space is a single space.
+     *
+     * This is equivalent to:
+     *      destination.append(source.replaceAll("\\s+", " "))
+     * but does not create a Pattern object that needs to compile the match
+     * string; it also prevents us from having to make a Matcher object as well.
+     *
+     */
+    public static void appendCompactedString(final StringBuilder destination, final String source) {
+        boolean previousIsWhiteSpace = false;
+        int length = source.length();
+
+        for (int i = 0; i < length; i++) {
+            char ch = source.charAt(i);
+            if (isWhiteSpace(ch)) {
+                if (previousIsWhiteSpace) {
+                    continue;
+                }
+                destination.append(CHAR_SPACE);
+                previousIsWhiteSpace = true;
+            } else {
+                destination.append(ch);
+                previousIsWhiteSpace = false;
+            }
+        }
+    }
+
+    /**
+     * Tests a char to see if is it whitespace.
+     * This method considers the same characters to be white
+     * space as the Pattern class does when matching \s
+     *
+     * @param ch the character to be tested
+     * @return true if the character is white  space, false otherwise.
+     */
+    private static boolean isWhiteSpace(final char ch) {
+        if (ch == CHAR_SPACE) {
+            return true;
+        }
+        if (ch == CHAR_TAB) {
+            return true;
+        }
+        if (ch == CHAR_NEW_LINE) {
+            return true;
+        }
+        if (ch == CHAR_VERTICAL_TAB) {
+            return true;
+        }
+        if (ch == CHAR_CARRIAGE_RETURN) {
+            return true;
+        }
+        if (ch == CHAR_FORM_FEED) {
+            return true;
+        }
+        return false;
+    }
+
+    public static String fromInteger(Integer value) {
+        return Integer.toString(value);
+    }
+
+    public static String fromLong(Long value) {
+        return Long.toString(value);
+    }
+
+    public static String fromString(String value) {
+        return value;
+    }
+
+    public static String fromBoolean(Boolean value) {
+        return Boolean.toString(value);
+    }
+
+    public static String fromFloat(Float value) {
+        return Float.toString(value);
+    }
```

Comment:
These from* methods (above and below) feel a bit weird and aren't used in many places. Maybe we could inline them in-place where they're used instead of migrating them?

Reply:
👍 generally - I agree - the only place I think this potentially does add value is in things like `fromInstant` where it'll (hopefully) force us to use a consistent format. For anything that has a `[type]#toString` method we should just use that.

Reply:
ValueToStringConverters uses StringUtils.fromFloat.    I'm not familiar with the inner workings of ValueToStringConverters.  Can I leave StringUtils.fromFloat as is?